src/kdbindings/connection_evaluator.h

```diff
@@ -67,14 +69,31 @@ class ConnectionEvaluator
         m_deferredSlotInvocations.clear();
     }
 
+protected:
+    /**
+     * @brief Called when a new slot invocation is added.
+     *
+     * This function can be overwritten by subclasses to get notified whenever a new invocation is added to this evaluator.
+     * The default implementation does nothing and does not have to be called by subclasses when overriding.
+     *
+     * ⚠️  *Note that this function will be executed on the thread that added the connection (i.e. the thread that emitted the signal),
```

Comment:
```suggestion
     * ⚠️  *Note that this function will be executed on the thread that enqueued the slot invocations (i.e. the thread that called .emit() on the signal),
```

I'd change the wording here as "that added the connection" could potentially be read as "that called .connect()". It's explained in the parentheses immediately, but I think that for casual user, "adding connection" means actually doing `.connect()`

Also, if you feel like it, you may add the explanation I added:
```
For example, if you plan to evaluate (execute) the slot invocations in some "main" thread A
and a signal is emitted in thread B, than this method is a good place to "wake up" the event
loop of thread A so it could call `evaluateDeferredConnections()` .
```

Reply:
I have incorporated the wording changes and additionally added your example (in a slightly tweaked form) to the description.
Added you as a co-author as well :)

Reply:
And I've removed the ConnectionHandle& as a parameter. You mentioned that you didn't need it in your fork, so let's not expose more information than is necessary.